File: backend/internal/models/contact_message.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type ContactMessage struct {
	ID        uuid.UUID `gorm:"type:uuid;primaryKey;default:gen_random_uuid()" json:"id"`
	Name      string    `gorm:"type:varchar(255);not null" json:"name"`
	Email     string    `gorm:"type:varchar(255);not null" json:"email"`
	Subject   *string   `gorm:"type:varchar(255)" json:"subject"`
	Message   string    `gorm:"type:text;not null" json:"message"`
	IsRead    bool      `gorm:"default:false" json:"isRead"`
	IPAddress *string   `gorm:"type:varchar(45)" json:"-"`
	CreatedAt time.Time `gorm:"not null;default:now()" json:"createdAt"`
}

File: frontend/app/types/contact-message.ts
export interface ContactMessage {
  id: string
  name: string
  email: string
  subject?: string | null
  message: string
  isRead: boolean
  createdAt: string
}

export interface RawContactMessage {
  id: string
  name: string
  email: string
  subject?: string | null
  message: string
  is_read?: boolean
  isRead?: boolean
  created_at?: string
  createdAt?: string
}

export function normalizeContactMessage(raw: RawContactMessage): ContactMessage {
  return {
    id: raw.id,
    name: raw.name,
    email: raw.email,
    subject: raw.subject ?? null,
    message: raw.message,
    isRead: raw.isRead ?? raw.is_read ?? false,
    createdAt: raw.createdAt ?? raw.created_at ?? '',
  }
}

export interface ContactFormState {
  name: string
  email: string
  subject: string
  message: string
  isSubmitting: boolean
  isSuccess: boolean
  error: string | null
}
